File: app/src/main/java/com/jmalinen/blackjack/model/CasinoRules.kt
package com.jmalinen.blackjack.model

data class CasinoRules(
    val numberOfDecks: Int = 6,
    val dealerStandsOnSoft17: Boolean = true,
    val dealerPeeks: Boolean = true,
    val blackjackPayout: BlackjackPayout = BlackjackPayout.THREE_TO_TWO,
    val surrenderPolicy: SurrenderPolicy = SurrenderPolicy.NONE,
    val doubleAfterSplit: Boolean = true,
    val resplitAces: Boolean = false,
    val maxSplitHands: Int = 4,
    val hitSplitAces: Boolean = false,
    val doubleOnAnyTwo: Boolean = true,
    val insuranceAvailable: Boolean = true,
    val threeSevensPays3to1: Boolean = false,
    val initialChips: Int = 1000,
    val minimumBet: Int = 10,
    val maximumBet: Int = 500
)

enum class BlackjackPayout(val multiplier: Float, val displayName: String) {
    THREE_TO_TWO(1.5f, "3:2"),
    SIX_TO_FIVE(1.2f, "6:5")
}

enum class SurrenderPolicy(val displayName: String) {
    NONE("No Surrender"),
    LATE("Late Surrender"),
    EARLY("Early Surrender")
}

File: ios/Blackjack/Blackjack/Model/CasinoRules.swift
struct CasinoRules: Equatable {
    var numberOfDecks: Int = 6
    var dealerStandsOnSoft17: Bool = true
    var dealerPeeks: Bool = true
    var blackjackPayout: BlackjackPayout = .threeToTwo
    var surrenderPolicy: SurrenderPolicy = .none
    var doubleAfterSplit: Bool = true
    var resplitAces: Bool = false
    var maxSplitHands: Int = 4
    var hitSplitAces: Bool = false
    var doubleOnAnyTwo: Bool = true
    var insuranceAvailable: Bool = true
    var threeSevensPays3to1: Bool = false
    var initialChips: Int = 1000
    var minimumBet: Int = 10
    var maximumBet: Int = 500
    var trainSoftHands: Bool = false
    var trainPairedHands: Bool = false
    var extraPlayers: Int = 0

    var isTrainingMode: Bool { trainSoftHands || trainPairedHands }
}

enum BlackjackPayout: Equatable {
    case threeToTwo
    case sixToFive

    var multiplier: Float {
        switch self {
        case .threeToTwo: 1.5
        case .sixToFive: 1.2
        }
    }

    var displayName: String {
        switch self {
        case .threeToTwo: "3:2"
        case .sixToFive: "6:5"
        }
    }
}

enum SurrenderPolicy: Equatable {
    case none
    case late
    case early

    var displayName: String {
        switch self {
        case .none: "No Surrender"
        case .late: "Late Surrender"
        case .early: "Early Surrender"
        }
    }
}
